[FIX] base_import_match: "Unable to use a closed cursor"

zaoral opened the pull request:
Cached methods should never return recordsets, which are tied to specific context, env, cursor, uid. Instead, they should return IDs which, later, can be browsed in the current context.

With this change _usable_rules() cached method is now returing ids instead of a recordset, and also the _match_find method is properly browsing the results ids of the cached method.

zaoral committed:
[FIX] base_import_match: "Unable to use a closed cursor"

Cached methods should never return recordsets, which are tied to
specific context, env, cursor, uid. Instead, they should return IDs
which, later, can be browsed in the current context.

With this change _usable_rules() cached method is now returing ids instead
of a recordset, and also the _match_find method is properly browsing the
results ids of the cached method.

pedrobaeza reviewed (approved):
One note: the parameters of the cached method can be recordsets. It's the returned result which can't be.

zaoral commented:
Hi @pedrobaeza  ! thank you for your review. I have updated the PR description to avoid confusion, also updated the commit message taking into account what you have mentioned. 

Best Regards

yajo commented:
/ocabot merge patch
